### ISSUE TITLE
Uncaught TypeError: Cannot add property loaded, object is not extensible

### ISSUE THREAD

**Issue by andreisocaciu** [silvia-odwyer/lightbox.js#30]
I get this error after the image loads.

```
Uncaught TypeError: Cannot add property loaded, object is not extensible
    at setItemLoaded (index.modern.js:474:1)
    at onLoad (index.modern.js:639:1)
```

Project with NextJS
```
next: 13.4.19
```

**Comment by silvia-odwyer** [comment]
@andreisocaciu Hi Andrei, Thanks very much for letting me know about this error. 

I'm just wondering if you could you provide a code sample, so that I can take a look into the issue? I'll debug this and implement a fix for this bug.

**Comment by andreisocaciu** [comment]
I manage to isolate the issue to the `images` prop being an `immer` object – I use [redux-toolkit](https://redux-toolkit.js.org/) for global state – which freezes the state object so no mutation is possible.

From the [React Docs](https://react.dev/learn/updating-objects-in-state#:~:text=However%2C%20although%20objects%20in%20React%20state%20are%20technically%20mutable%2C%20you%20should%20treat%20them%20as%20if%20they%20were%20immutable%E2%80%94like%20numbers%2C%20booleans%2C%20and%20strings.%20Instead%20of%20mutating%20them%2C%20you%20should%20always%20replace%20them.):

> However, although objects in React state are technically mutable, you should treat them as if they were immutable—like numbers, booleans, and strings. Instead of mutating them, you should always replace them.

In my case, the state object is NOT mutable, hence the error.

Codesandbox: https://codesandbox.io/p/sandbox/lightbox-issue-sx7yxx

**Comment by silvia-odwyer** [comment]
@andreisocaciu Hi Andrei,
Thanks for providing the CodeSandbox, it's much appreciated! I've just updated the library so that the `images` prop is no longer mutated and is replaced with a new object instead. If you re-install the latest version of the library, could you let me know if it fixes the issue?

**Comment by andreisocaciu** [comment]
It's fixed in `0.7.6`. Thanks!
